internal/identity/identity.go

```diff
@@ -0,0 +1,297 @@
+			continue
+		}
+		if len(iface.HardwareAddr) == 0 {
+			continue
+		}
+		return iface.HardwareAddr.String()
+	}
+
+	return ""
+}
+
+// resolveGeneratedID returns a persisted UUID stored at
+// ~/.aeroarc/agent-id, creating it if necessary. Errors are ignored and
+// result in an empty string.
+func resolveGeneratedID(env resolverEnv) string {
+	home, err := env.userHomeDir()
+	if err != nil || home == "" {
+		return ""
+	}
+
+	path := filepath.Join(home, ".aeroarc", "agent-id")
+
+	// Attempt to load an existing ID.
+	if data, err := env.readFile(path); err == nil {
+		if id := normalizeID(string(data)); id != "" {
+			return id
+		}
+	}
+
+	// Generate a fresh UUIDv4.
+	uuid, err := generateUUIDv4()
+	if err != nil {
+		return ""
+	}
+
+	dir := filepath.Dir(path)
+	if dir != "" {
+		_ = env.mkdirAll(dir, 0o700)
+	}
+
+	_ = env.writeFile(path, []byte(uuid), 0o600)
+
+	return uuid
+}
+
+// readDeviceTreeSerial reads a device-tree serial-number file, stripping
+// trailing NUL terminators and whitespace as required on ARM platforms.
+func readDeviceTreeSerial(env resolverEnv, path string) string {
+	return readIDFromFile(env, path, true)
+}
+
+// readRaspberryPiSerial parses the CPU serial from /proc/cpuinfo on
+// Raspberry Pi systems.
+func readRaspberryPiSerial(env resolverEnv) string {
+	data, err := env.readFile("/proc/cpuinfo")
+	if err != nil {
+		return ""
+	}
+
+	lines := strings.Split(string(data), "\n")
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if !strings.HasPrefix(line, "Serial") {
+			continue
+		}
+
+		// Format is typically: "Serial  : 00000000abcdef01"
+		parts := strings.SplitN(line, ":", 2)
+		if len(parts) != 2 {
+			continue
+		}
+
+		serial := normalizeID(parts[1])
+		if serial != "" {
+			return serial
+		}
+	}
+
+	return ""
+}
+
+// readIDFromFile reads the contents of a file and returns a normalized
+// identifier string, optionally stripping NUL bytes which are common in
+// device-tree blobs.
+func readIDFromFile(env resolverEnv, path string, stripNull bool) string {
+	data, err := env.readFile(path)
+	if err != nil {
+		return ""
+	}
+	if stripNull {
+		// Device-tree files often contain NUL padding; strip all NUL bytes.
+		data = bytes.ReplaceAll(data, []byte{0x00}, nil)
+	}
+
+	return normalizeID(string(data))
+}
+
+// normalizeID trims whitespace and NUL characters from the provided string.
+func normalizeID(s string) string {
+	// Remove NUL from both ends, then whitespace.
+	s = strings.Trim(s, "\x00")
+	s = strings.TrimSpace(s)
+	return s
+}
+
+// computeFinalID constructs the composite identity digest as:
+// sha256(hardwareID + machineID + macAddress + generatedID), hex-encoded.
+func computeFinalID(hardwareID, machineID, macAddress, generatedID string) string {
+	h := sha256.New()
+	_, _ = h.Write([]byte(hardwareID))
+	_, _ = h.Write([]byte(machineID))
+	_, _ = h.Write([]byte(macAddress))
+	_, _ = h.Write([]byte(generatedID))
+	sum := h.Sum(nil)
+	return hex.EncodeToString(sum)
```

Comment:
should we write the finalID to the agent-id file in like a .aero-arc/id file? Also mac address can change no? If so, that would change the hashed value